send-transaction-service/src/send_transaction_service.rs

```diff
@@ -128,6 +128,54 @@ impl Default for Config {
     }
 }
 
+/// A struct responsible for providing the up-to-date leader information
+/// used for sending transactions.
+pub const LEADER_INFO_REFRESH_RATE_IN_MS: u64 = 1000;
```

Comment:
i think there are some `1000.min(...)` elsewhere in this file that are no longer needed

Reply:
I think it would not impact logic on receive transaction. But it can impact retry timing. Supposing the retry interval is 10 seconds, and transactions are evenly added into the queue in time. The 1 second largest value can help processing the transactions in smaller batches instead of processing larger batch of transactions all in the same intervals. So it will process 10% of transactions being retried every 1 second instead of sleeping whole 10 seconds doing nothing and then at 10th second retry all of them. I would say it will do the retries more evenly probably better take advantage of the bandwidth. I think we could leave this in-place.

Reply:
oh. replace 1000 with LEADER_INFO_REFRESH_RATE_IN_MS?

Reply:
that'd work. or a shorter named temporary assigned from it, if that makes sense in context

Reply:
Got it. Corrected as appropriate. I removed it from receive_txn_thread as it is useless. And defined a variable for retry_thread.